change/@ni-nimble-components-c4841b4a-c42f-4d23-9748-857bbff400dd.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Support dynamic arbitrary header content",
+  "packageName": "@ni/nimble-components",
```

Comment:
It would be good to think about how a client would style the content. Some questions include:
1. I assume they can just style the content in their own stylesheet since it's slotted? Any implications around uniquely identifying which column is being styled? Perhaps this is another use case for `column-id` (or just `id).
2. What guarantees do we provide about how the table will lay out the header content? This came up in the context of potentially wanting numeric columns to have right-aligned column headers (decision still pending). While in that case it would likely be the custom column developer that wants to influence header layout, I could imagine clients (users of columns) wanting to configure where the header content goes too. We could choose not to let them configure this to start, but we should consider whatever behavior we provide to be part of our contract.
3. Related: what if a user provides content that's taller than other column headers? Do we grow all the headers? What if it's wider than the fixed width? Do we always use `overflow: hidden`?

Once we have decisions on these we should document/demonstrate them in some combination of:
1. the spec
2. storybook API docs
3. storybook matrix tests
4. example apps

Reply:
I've had off-line conversations with both @jattasNI and @mollykreis regarding this, and at the moment I think it's prudent to table this discussion insofar as it relates to this PR. The reason behind this is that it's currently unclear whether we should be providing _any_ formal mechanism for a user to configure styling for header content. Particularly since it is already within the client's ability to do something like the following:

```
<nimble-table>
    <nimble-table-column-text><span style="color: red;">Colored Header</span></nimble-table-column-text>
</nimble-table>
```

This approach, admittedly, doesn't offer complete control over styling the color of something like an icon, but it may be that we don't want to promote such a thing, and instead have it be limited to how icons can be colored in general via their `severity` attribute.

Alignment of the content is also strange in that, at the moment, we don't have a clear UX decision/vision on how headers should be aligned with respect to their data, particularly when considering the other elements that will ultimately make up the header content (the menu dropdown, the sort direction glyph, etc...).

Reply:
Sounds good. As we discussed, that means our official policy today is "Nimble will style your text or icon content for you. If you need custom styles, reach out to us to discuss the use case". We should document this once we have a place to document column configuration.